File: 백준/Silver/10814. 나이순 정렬/나이순 정렬.py
import sys
input = sys.stdin.readline

N = int(input())
arr = []
for k in range(1, N+1):
    age, name = input().split()
    arr.append([int(age), name, k])

arr.sort(key=lambda x:(x[0],x[2],x[1]))

for person in arr:
    person.pop()
    print(*person)

File: 백준/Silver/10814. 나이순 정렬/나이순 정렬.js
function solution(input) {
  const [_, ...arrs] = input;
  const orders = {};

  const data = arrs.map((arr, i) => {
    orders[arr] = i;
    const _arr = arr.split(" ");
    _arr[0] = +_arr[0];
    return _arr;
  });

  const result = data.sort((a, b) => {
    const [age1, name1] = a;
    const [age2, name2] = b;
    return age1 === age2 ? orders[name1] - orders[name2] : age1 - age2;
  });

  return result.map((arr) => arr.join(" "));
}

(function main() {
  let isLocal = false;
  let fs = require("fs");
  let filePath = isLocal ? "t.txt" : "/dev/stdin";
  let input = fs.readFileSync(filePath).toString().trim().split("\n");

  const answer = solution(input).join("\n");
  console.log(answer);
})();